src/org/thoughtcrime/securesms/MessageSelectorFragment.java

```diff
@@ -0,0 +1,79 @@
+package org.thoughtcrime.securesms;
+
+import android.Manifest;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.text.util.Linkify;
+import android.widget.TextView;
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.view.ActionMode;
+import androidx.fragment.app.Fragment;
+
+import com.b44t.messenger.DcMsg;
+
+import org.thoughtcrime.securesms.connect.ApplicationDcContext;
+import org.thoughtcrime.securesms.permissions.Permissions;
+import org.thoughtcrime.securesms.util.SaveAttachmentTask;
+
+import java.util.Set;
+
+public abstract class MessageSelectorFragment extends Fragment
+{
+  protected ActionMode actionMode;
+  protected ApplicationDcContext dcContext;
+
+  protected void handleDisplayDetails(DcMsg dcMsg) {
+    String infoStr = dcContext.getMsgInfo(dcMsg.getId());
+    AlertDialog d = new AlertDialog.Builder(getActivity())
+            .setMessage(infoStr)
+            .setPositiveButton(android.R.string.ok, null)
+            .create();
+    d.show();
+    try {
+      //noinspection ConstantConditions
+      Linkify.addLinks((TextView) d.findViewById(android.R.id.message), Linkify.WEB_URLS);
+    } catch(NullPointerException e) {
+      e.printStackTrace();
+    }
+  }
+
+  protected void handleDeleteMessages(final Set<DcMsg> messageRecords) {
+    int messagesCount = messageRecords.size();
+
+    new AlertDialog.Builder(getActivity())
+            .setMessage(getActivity().getResources().getQuantityString(R.plurals.ask_delete_messages, messagesCount, messagesCount))
+            .setCancelable(true)
+            .setPositiveButton(R.string.delete, (dialog, which) -> {
+                int[] ids = DcMsg.msgSetToIds(messageRecords);
+                dcContext.deleteMsgs(ids);
+                actionMode.finish();
```

Comment:
```suggestion
                if (actionMode != null) actionMode.finish();
```
On my phone, this crashed without this line